packages/core/src/components/forms/_input.scss

```diff
@@ -29,6 +29,35 @@ Styleguide input
     @include pt-input-large();
   }
 
+  &.#{$ns}-small {
+    .#{$ns}-button {
+      @include pt-button-height($pt-button-height-smaller);
+      margin: ($pt-input-height-small - $pt-button-height-smaller) / 2;
+    }
+
+    .#{$ns}-tag {
+      @include pt-button-height($pt-button-height-smaller);
+      margin: ($pt-input-height-small - $pt-button-height-smaller) / 2;
+    }
+
+    > .#{$ns}-icon,
+    .#{$ns}-input-action > .#{$ns}-spinner {
+      margin: ($pt-input-height-small - $pt-icon-size-standard) / 2;
+    }
+
+    .#{$ns}-input {
+      @include pt-input-small();
+
+      &:not(:first-child) {
+        padding-left: $pt-button-height-smaller;
+      }
+
+      &:not(:last-child) {
```

Comment:
these are not necessary because inputs never have children. padding should always be set directly. refer to other input styles for inspiration.